### PR TITLE
feat(compiler): add config to allow external pipeline updates

### PR DESCRIPTION
**Pull request by JordanSussman**
Initial implementation for go-vela/community#88.

## local testing

1. update the `docker-compose` file to look like:

```yaml
# Copyright (c) 2020 Target Brands, Inc. All rights reserved.
#
# Use of this source code is governed by the LICENSE file in this repository.

version: '3'

services:

  server:
    container_name: server
    build:
      context: .
      dockerfile: Dockerfile
    image: vela:local
    networks:
      - vela
    environment:
      VELA_ADDR: 'http://localhost:8080'
      VELA_WEBUI_ADDR: 'http://ui:80'
      VELA_DATABASE_DRIVER: postgres
      VELA_DATABASE_CONFIG: postgres://vela:zB7mrKDTZqNeNTD8z47yG4DHywspAh@postgres:5432/vela?sslmode=disable
      VELA_LOG_LEVEL: trace
      VELA_QUEUE_DRIVER: redis
      VELA_QUEUE_CONFIG: redis://redis:6379
      VELA_QUEUE_WORKER_ROUTES: large,small,docker,large:docker,small:docker
      VELA_PORT: ":8080"
      VELA_SECRET: zB7mrKDTZqNeNTD8z47yG4DHywspAh
      VELA_SOURCE_DRIVER: github
      VELA_SECRET_VAULT: "true"
      VELA_SECRET_VAULT_ADDR: http://vault:8200
      VELA_SECRET_VAULT_TOKEN: vela
      VELA_MODIFICATION_ADDR: "http://modification:8080/config"
      VELA_MODIFICATION_SECRET: "217758e80e8db1b597d2c52740a35580"
    env_file:
      - secrets.env
    restart: always
    ports:
      - "8080:8080"
    depends_on:
      - postgres
      - redis
      - vault

  worker:
    container_name: worker
    image: target/vela-worker
    networks:
      - vela
    environment:
      VELA_BUILD_LIMIT: 1
      VELA_BUILD_TIMEOUT: 30m
      VELA_LOG_LEVEL: trace
      VELA_QUEUE_DRIVER: redis
      VELA_QUEUE_CONFIG: redis://redis:6379
      VELA_QUEUE_WORKER_ROUTES: large,docker,large:docker
      VELA_RUNTIME_DRIVER: docker
      VELA_SERVER_ADDR: http://server:8080
      VELA_SERVER_SECRET: zB7mrKDTZqNeNTD8z47yG4DHywspAh
    restart: always
    ports:
      - "8081:8080"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
    depends_on:
      - server

  ui:
    container_name: ui
    image: target/vela-ui
    networks:
      - vela
    ports:
      - "8888:80"
    environment:
      VELA_API: "http://localhost:8080"
    depends_on:
      - server

  redis:
    container_name: redis
    image: redis:5-alpine
    networks:
      - vela
    ports:
      - "6379:6379"

  postgres:
    container_name: postgres
    image: postgres:12-alpine
    networks:
      - vela
    environment:
      POSTGRES_DB: vela
      POSTGRES_PASSWORD: zB7mrKDTZqNeNTD8z47yG4DHywspAh
      POSTGRES_USER: vela
    ports:
      - "5432:5432"

  vault:
    image: vault:0.9.6
    container_name: vault
    command: server -dev
    networks:
      - vela
    environment:
      VAULT_DEV_LISTEN_ADDRESS: 0.0.0.0:8200
      VAULT_DEV_ROOT_TOKEN_ID: vela
    ports:
      - "8200:8200"
    cap_add:
      - IPC_LOCK

  modification:
    image: jsussman/vela-modification:0.1.0
    container_name: modification
    networks:
      - vela
    ports:
      - "8100:8080"

networks:
  vela:
```

2. build the binary - `make build`
3. stand up the compose containers - `docker-compose up --build`
4. send payload to server to initiate a build
5. take notice of how a step named `modification` is appended to the list of executed steps

The code for the `modification` container is viewable [here](https://github.com/JordanSussman/mod-sample/blob/master/main.go).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,10 @@ module github.com/go-vela/server
 
 go 1.13
 
+replace github.com/go-vela/compiler => github.com/JordanSussman/compiler v0.1.3-0.20200817234448-4421bb944025
```
Comment (JordanSussman) [inline go.mod:5]: Will remove once https://github.com/go-vela/compiler/pull/81 is merged.

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,10 @@ module github.com/go-vela/server
...
+
+replace github.com/go-vela/types => github.com/JordanSussman/types v0.1.2-0.20200817233341-2853ce2956de
```
Comment (JordanSussman) [inline go.mod:7]: Will remove once https://github.com/go-vela/types/pull/100 is merged.

File: cmd/vela-server/main.go [hunk 3]
```diff
@@ -175,6 +175,23 @@ func main() {
 			Usage:   "source commit status context",
 			Value:   "continuous-integration/vela",
 		},
+
+		&cli.StringFlag{
```
Comment (kneal) [inline cmd/vela-server/main.go:179]: Would these flags ever need to be arrays? I'm trying to understand if an administrator would ever want to maintain more than one modification system.

Comment (JordanSussman) [reply cmd/vela-server/main.go:179]: I'd opt to only allow a single modification system until we determine the use case for multiple. I see a potential use case for having different types of modification systems that are called at different points of compilation, but that is likely outside the scope of the initial implement ion IMO.

**Review by meinenec — APPROVED** [review]

**Pull request merged**